mixer.cpp

```diff
@@ -51,11 +51,12 @@ struct MixerChannel {
 
 	void mix(int16_t &sample) {
 		if (_data) {
-			const uint32_t pos = _pos.getInt();
+			uint32_t pos = _pos.getInt();
 			_pos.offset += _pos.inc;
 			if (_loopLen != 0) {
 				if (pos >= _loopPos + _loopLen) {
-					_pos.offset = _loopPos << Frac::BITS;
+					pos = _loopPos;
+					_pos.offset = (_loopPos << Frac::BITS) + _pos.inc;
```

Comment:
Because this value will be used in the next iteration (`uint32_t pos = _pos.getInt();`). If _pos.inc isn't added, then the position _loopPos would be used twice.

Reply:
Good catch.

Reply:
SfxPlayer.mixChannel could probably need a similar change...